### PR TITLE
XS✔ ◾ [docs] add secrets management 

### PR DESCRIPTION
**Pull request by mustafaStakater**
❌ **Add a description.**

### CODE REVIEW

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-1c5bb8e4`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-bfee952a`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-0494ae60`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-fa939cd8`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-ffe1b523`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-46f6bc2a`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-49157d89`

**Comment by stakater-user** [comment]
@bilalbokharee Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-b8c3d8fe`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-3c56e881`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-dc438be6`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-8efa9179`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-ebc66bc9`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-48d3f6f2`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-2e712e42`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-a2773358`

**Comment by stakater-user** [comment]
@bilalbokharee Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-34c639c6`

**Comment by stakater-user** [comment]
@bilalbokharee Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-2ea51e63`

**Comment by stakater-user** [comment]
@bilalbokharee Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-6e096265`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-f550dcf2`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-27530a53`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-20da1a73`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-a7c22977`

**Comment by stakater-user** [comment]
@mustafaStakater Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-3b2af2dc`

**Comment by stakater-user** [comment]
@bilalbokharee Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-3b98842c`

**Comment by stakater-user** [comment]
@bilalbokharee Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-3a545358`

**Comment by stakater-user** [comment]
@bilalbokharee Image is available for testing. `docker pull stakater/workshop-exercise:SNAPSHOT-PR-149-720ec540`

File: docs/6-secret-management-in-saap/6-secret-management.md [hunk 1]
```diff
@@ -1,18 +1,337 @@
...
+        template:
+          type: kubernetes.io/basic-auth
+          metadata:
+            annotations:
+              tekton.dev/git-0: 'https://gitlab.apps.devtest.vxdqgl7u.kubeapp.cloud'
+      dataFrom:
+        - key: nordmart-review-ui-page-title
+
+   ```
+   3. External Secrets Operator (ESO) watches for ExternalSecret CR creation.
```
Comment (mustafaStakater) [inline docs/6-secret-management-in-saap/6-secret-management.md:241]: combine 3 and 4

File: docs/6-secret-management-in-saap/6-secret-management.md [hunk 2]
```diff
@@ -1,18 +1,337 @@
...
+      ```
+         dataFrom:
+            - key: gitlab-pat
+      ```
+
+   7. Vault returns the requested secret data to External Secrets Operator (ESO).
+
+   8. External Secrets Operator (ESO) creates a Kubernetes Secret. 
+
+   9. Kubernetes Secret is consumed by Application.
```
Comment (mustafaStakater) [inline docs/6-secret-management-in-saap/6-secret-management.md:264]: add step for 
Adding the secret in application values.yaml by specifying `deployment.env.PAGE_TITLE` with valueFrom field referencing this secret

add step for `helm template deploy/ -n <TENANT>-dev | oc apply -f -`

add step for showing the updated ui

File: docs/6-secret-management-in-saap/6-secret-management.md [hunk 3]
```diff
@@ -1,18 +1,337 @@
...
+
+      ```
+         refreshInterval: "1m"
+      ```
+
+   3. External Secrets Operator (ESO) receives the update and starts reconciliation.
+
+   4. External Secrets Operator (ESO) updates the values of Kubernetes Secret with the new values user just added in Vault.
+
+   5. Stakater Reloader is continously watching the Kubernetes Secret for change.
```
Comment (mustafaStakater) [inline docs/6-secret-management-in-saap/6-secret-management.md:301]: Do add screenshot for secret being updated & pod getting restarted

File: docs/6-secret-management-in-saap/6-secret-management.md [hunk 4]
```diff
@@ -1,18 +1,337 @@
...
+
+   6. It receives the update instantly.
+
+   7. Stakater Reloader performs a rolling upgrade on Kubernetes resource(s). Application is up again with the updated secret values in no time!
```
Comment (mustafaStakater) [inline docs/6-secret-management-in-saap/6-secret-management.md:305]: 3. Add step with the updated ui using the secret

**Review by AsfaMumtaz — APPROVED** [review]

**Pull request merged**